### ISSUE TITLE
Side nav focus is inconsistent and keyboard navigation doesn't work as expected

### ISSUE THREAD

**Issue by alisonjoseph** [carbon-design-system/carbon-platform#1302]
**Bug🐛:** side nav focus is inconsistent and keyboard navigation doesn't work as expected
**Page:** any page with a sidenav
**Browser:** Chrome
**Suggestion:** the `SideNavLink` focus is missing a left border, however, `SideNavMenu` has a full border. Additonally, if you tab into it, sometimes it focuses on an inner element, and not all items are tabbable, but they are arrow key navegable (seems like maybe tab isn't intended). `SideNavMenu` are also not expanding with keyboard enter/space

https://user-images.githubusercontent.com/32556167/191101767-e0ff0dfa-f306-44da-a117-57fd2472b13b.mov

_Originally posted by @jnm2377 in https://github.com/carbon-design-system/carbon-platform/discussions/1268#discussioncomment-3684148_

**Comment by alisonjoseph** [comment]
I believe keyboard interactions are working as intended, as this uses treeview (arrows not tabs, this allows the user to tab out of the nav vs. having to go through it), other than enter not working to navigate to a page. (tracked https://github.com/carbon-design-system/carbon-platform/issues/1291) 

Need to fix focus state being offset to the left / missing left border

**Comment by alisonjoseph** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Development-624c3fc8b7972e001130fef7/issues/carbon-design-system/carbon-platform/1302?planning-poker) with Zenhub @andreancardona @francinelucca @jdharvey-ibm
